shared/src/main/java/com/couchbase/lite/Dictionary.java

```diff
@@ -103,9 +103,9 @@ public Dictionary set(Map<String, Object> dictionary) {
     @Override
     public List<String> getKeys() {
         if (!changed)
-            return super.getKeys();
+            return new ArrayList(super.getKeys());
```

Comment:
There is no needs to create a new ArrayList here as super.getKeys() already returns a copy.